modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/index.federation.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import process from 'process';
+
+export * as __classnames from 'classnames';
+export * as __formik from 'formik';
+export {default as __propTypes} from 'prop-types';
+export {default as __react} from 'react';
+export * as __reactDnd from 'react-dnd';
+export * as __reactDndHtml5Backend from 'react-dnd-html5-backend';
+export {default as __reactDom} from 'react-dom';
+
+export {default as render} from './render.es';
+export {default as useEventListener} from './hooks/useEventListener.es';
+export {default as useInterval} from './hooks/useInterval.es';
+export {default as useIsMounted} from './hooks/useIsMounted.es';
+export {default as usePrevious} from './hooks/usePrevious.es';
+export {default as useStateSafe} from './hooks/useStateSafe.es';
+export {default as useThunk} from './hooks/useThunk.es';
+export {default as useTimeout} from './hooks/useTimeout.es';
+
+// Egregious hack because react-dnd expects `window.process` to exist:
+//
+// https://github.com/react-dnd/asap/blob/b6bebeb734/src/node/asap.ts#L24
```

Comment:
Oh god, this caused me no end of grief with previous react-dnd updates. I can't remember how I got around it though right now... The answer is probably somewhere [in here](https://github.com/liferay-frontend/liferay-portal/pull/106):

> -  Configure `frontend-js-react-web` to get `process` from `frontend-js-node-shims`; this is because — the horror 😱 — `@react-dnd/asap` [expects it to be a global](https://github.com/react-dnd/asap/blob/b6bebeb7342/src/node/asap.ts#L24).

Reply:
Ha, I see these lines are literally the ones I was thinking of. I should have known when I saw the word "egregious". 😂

Reply:
So we don't need `frontend-js-node-shims` any more because webpack takes care of it.

@jbalsas  Aren't you happy of this :point_up: ?

Reply:
> So we don't need frontend-js-node-shims any more because webpack takes care of it.

I already thought we didn't need them 😂